RecoTauTag/RecoTau/src/RecoTauCommonUtilities.cc

```diff
@@ -111,6 +111,7 @@ namespace reco {
       if (pfCand)
         return pfCand->positionAtECALEntrance();
 
+      assert(part);
```

Comment:
As far as I understand it it is protection against nullptr, right? It is unlikely to happen, but it is OK to have this protection. But, would it be more logical to move this assert to l.110 before dynamic_cast?

Reply:
Is it "unlikely to happen" or "should never happen"? If it is the latter, how about changing the function to take the `part` as const reference instead of pointer to const?

Reply:
I think both :) I think nobody have touched this code for years, but I think it is called with well behaving, i.e. non null pointers. Anyway I agree that using const ref is better design than what is present currently. I can put this to my todo, butI think that in scope of this PR it makes sense to move the assert a few lines earlier.

Reply:
sure, I will move it up

Reply:
done